flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/WindowEmitStrategy.scala

```diff
@@ -63,7 +63,7 @@ class WindowEmitStrategy(
 
   def produceUpdates: Boolean = {
     if (isEventTime) {
-      allowLateness > 0 || earlyFireDelayEnabled || lateFireDelayEnabled
+      earlyFireDelayEnabled || lateFireDelayEnabled
```

Comment:
If we only set allowLateness, then it won't work?

Reply:
Yes. I think this is a bug when we merging blink code. A window aggregation can produce updates only when early fire or late fire is enabled. `allowLateness` is just a hint to the statebackend for how long to keep the state when window is closed. If late data never trigger emission, the window still produce insert-only messages.

Reply:
Then we should modify `WindowOperatorBuilder.withAllowedLateness()` too?

Reply:
It's better we fix this in another issue if it's a bug

Reply:
I will still keep this fix in this PR to make the whole tests to pass. But I will split it into another commit.

Reply:
please add a test to verify this bug